### PR TITLE
✅ Test AWS2 SQS in isolation

### LINKED ISSUE #2606: Test AWS2 SQS in isolation
We currently have a test that combines SNS and SQS. This is because SNS is a bare pipe that needs to be wired to some other AWS service (SQS, S3 or maybe some other) to be able to ensure that a message has passed it.
Regardless of that an isolated SQS test should be possible and we should try to create one.

### PR DESCRIPTION
**Pull request by zbendhiba**
Fixes #2606

I didn't invent any of these tests. I've just split the tests to have 

- a project isolating the SQS tests
- a project for SNS tests, with only the SQS bits needed to perform those tests

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by zbendhiba** [comment]
Let me fix the build.

File: integration-test-groups/aws2/aws2-sqs/src/test/java/org/apache/camel/quarkus/component/aws2/sqs/it/Aws2SqsTest.java [hunk 1]
```diff
@@ -160,35 +160,35 @@ public void sqsAutoCreateDelayedQueue() {
...
     private void createDelayQueueAndVerifyExistence(String queueName, int delay) {
-        RestAssured.post("/aws2-sqs-sns/sqs/queue/autocreate/delayed/" + queueName + "/" + delay)
+        RestAssured.post("/aws2-sqs/queue/autocreate/delayed/" + queueName + "/" + delay)
                 .then()
                 .statusCode(200)
                 .extract()
                 .body()
                 .as(String[].class);
-        Awaitility.await().pollInterval(1, TimeUnit.SECONDS).atMost(120, TimeUnit.SECONDS).until(
+        Awaitility.await().pollInterval(1, TimeUnit.SECONDS).atMost(10, TimeUnit.SECONDS).until(
```
Comment (aldettinger) [inline integration-test-groups/aws2/aws2-sqs/src/test/java/org/apache/camel/quarkus/component/aws2/sqs/it/Aws2SqsTest.java:169]: It looks the sqs subset of sqs-sns to take far less time. What is the reason of narrowing down the timeout from 120 to 10 ? Do we save that much time ?

File: integration-test-groups/aws2-quarkus-client/aws2-sqs/pom.xml [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        <dependency>
+            <groupId>org.seleniumhq.selenium</groupId>
+            <artifactId>htmlunit-driver</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
```
Comment (aldettinger) [inline integration-test-groups/aws2-quarkus-client/aws2-sqs/pom.xml:81]: What is the purpose of those add-(test)-sources / group-sources sections below ? It doesn't ring any bells.

File: integration-test-groups/aws2/aws2-sqs/src/test/java/org/apache/camel/quarkus/component/aws2/sqs/it/Aws2SqsTest.java (hunk 1 above)
Comment (zbendhiba) [reply integration-test-groups/aws2/aws2-sqs/src/test/java/org/apache/camel/quarkus/component/aws2/sqs/it/Aws2SqsTest.java:169]: Oh, it was my local stuff. I need to put everything back as normal for tests against AWS.
Thanks for catching this

File: integration-test-groups/aws2-quarkus-client/aws2-sqs/pom.xml (hunk 2 above)
Comment (zbendhiba) [reply integration-test-groups/aws2-quarkus-client/aws2-sqs/pom.xml:81]: Basically, we create some projects that recopy some files from  the original AWS2 tests, by adding the Quarkus AWS clients. To make sure that the same projects run fine with a different pom.xml file.

Comment (zbendhiba) [reply integration-test-groups/aws2-quarkus-client/aws2-sqs/pom.xml:81]: Basically, we create some projects that recopy some files from the original AWS2 tests, by adding the Quarkus AWS clients. To make sure that the same projects run fine with a different pom.xml file.

File: integration-test-groups/aws2/aws2-sqs/src/test/java/org/apache/camel/quarkus/component/aws2/sqs/it/Aws2SqsTest.java (hunk 1 above)
Comment (zbendhiba) [reply integration-test-groups/aws2/aws2-sqs/src/test/java/org/apache/camel/quarkus/component/aws2/sqs/it/Aws2SqsTest.java]: fixed !!

File: integration-test-groups/aws2-quarkus-client/aws2-sqs/pom.xml (hunk 2 above)
Comment (aldettinger) [reply integration-test-groups/aws2-quarkus-client/aws2-sqs/pom.xml:81]: Ok, indeed it make sense now.

**Review by aldettinger — APPROVED** [review]

**Comment by zbendhiba** [comment]
There is going to be a conflict between this PR and this one https://github.com/apache/camel-quarkus/pull/3367
As the Quarkus AWS2 client has moved to  Quarkiverse.
Let's wait for the merge of previous one, and I'll rebase, correct this one

**Comment by zbendhiba** [comment]
Rebased from main.

**Pull request merged**